libraries/AP_GPS/AP_GPS_ERB.cpp

```diff
@@ -178,6 +178,8 @@ AP_GPS_ERB::_parse_gps(void)
         }
         state.num_sats = _buffer.stat.satellites;
         if (next_fix >= AP_GPS::GPS_OK_FIX_3D) {
+            // use the uart receive time to make packet timestamps more accurate
+            set_uart_timestamp(_payload_length + sizeof(erb_header) + 2);
```

Comment:
If MSG_STAT is sent at a slower rate than POS and VEL are, won't this be a problem? I mean the message will look it's older than it really is.
Also, it looks like it could use same strategy as Ublox.

Reply:
If so then it is an existing bug, as MSG_STAT is the only msg where we update state.last_gps_time_ms

Reply:
But the uart timestamp isn't used for last_gps_time_ms (it would actually make sense to also use the UART method below instead of the millis directly), neither is the time for POS and VEL taken from or compared to it.

Reply:
By the way, any doubt about the protocol, it's at https://files.emlid.com/ERB.pdf I'm sure Emlid guys are also available to give any explanation needed.